src/bacnet/basic/object/netport.c

```diff
@@ -723,14 +723,16 @@ uint8_t Network_Port_MAC_Address_Value(
 bool Network_Port_MAC_Address(
     uint32_t object_instance, BACNET_OCTET_STRING *mac_address)
 {
-    uint8_t mac_len = 0;
+    bool status = false;
 
     if (mac_address) {
-        mac_len = Network_Port_MAC_Address_Value(
-            object_instance, mac_address->value, sizeof(mac_address->value));
+        uint8_t mac_tmp[4 + 2] = { 0 };
+        uint8_t mac_len = Network_Port_MAC_Address_Value(object_instance, mac_tmp, sizeof(mac_tmp));
```

Comment:
if mac_len >0 then status = octetstring_init(...) ?

Reply:
Yeah but that returns true even though mac_len = 0. It only returns false if input is null or length is greater than MAX_OCTET_STRING_BYTES

Reply:
Since this function is already looking inside the BACNET_OCTET_STRING structure, fixing should be:
```
    if (mac_address) {
        mac_len = Network_Port_MAC_Address_Value(
            object_instance, mac_address->value, sizeof(mac_address->value));
        mac_address->length = mac_len;
    }
```